More UI Features

brschwalm opened the pull request:
- Added Simplex exchange to the Buy / Fund modal
- Added a fund wallet item to the Account Menu
- Added a high-value warning message
- Added a reminder to back up your account key

notes: 
- Still waiting on the icon for Fund Wallet, just used a placeholder for now
- I had to set the reminderDismissed flag from the importWalletForm, not the importWallet saga, as we discussed, because the importWallet saga is also used to load the wallet when the user enters the pin.  The flag should only get set the first time the user is actually importing their wallet.
- I had to put the Warnings / Reminder in the ScreenFrameWithFeed so it works correctly in Mobile as well as Desktop.
- I broke the warnings logic into a component, selector and hook.  Here's why: 
   - In order for the ScreenContentFrame to avoid unnecessary scrollbars, it has to know if there are warnings or not (so it can adjust the height).  
   - And since it's fairly expensive to figure out (read 4 values from state, convert 2 balances from wei, and logic), and since the content frame is always there, I thought a selector would be better than constantly re-evaluating it across multiple screens.  
   - The hook is because I also need to know the location.  
   - So in the end, both HomeScreenWarnings and ScreenContentFrame can just call the hook and they won't have to re-calculate unless something changes.

jmrossy commented on src/components/Notification.tsx:
```diff
@@ -6,41 +9,61 @@ interface NotificationProps {
+  useEffect(() => {
+    if (isDismissed && props.onDismiss) {
+      setTimeout(() => {
+        if (props.onDismiss) props.onDismiss()
```
Is the additional if check in here to protect against dismount? Should we return a clearTimeout instead? Maybe a custom useTimeout effect to handle this for us would be nice since we have this pattern in a few places. It could go next to useInterval in time.ts

brschwalm replied:
The additional check is because the typescript compiler can't determine that onDismiss is valid, despite the first if.  I imagine it's because the function execution happens inside a callback, so there's no guarantee.  I kept the first if there because there's no reason to set the timer if there's no onDismiss.  But yeah, I'll look at converting this to a useTimeout hook.

jmrossy commented on src/components/header/AccountMenu.tsx:
```diff
@@ -52,6 +64,10 @@ export const AccountMenu = () => {
         navigate('/change-pin')
         break
+      case 'fund':
+        setOpen(false)
```
We should probably have all these options setOpen(false) right? Should we move this down to after the switch?

jmrossy commented on src/components/layout/ScreenContentFrame.tsx:
```diff
@@ -10,9 +11,11 @@ interface FrameProps {
 
 export function ScreenContentFrame(props: PropsWithChildren<FrameProps>) {
   const { onClose } = props
+  const warnings = useHomeScreenWarnings()
+  const height = warnings ? { height: 'calc(100% - 5em)' } : { height: '100%' }
```
This approach is okay but to avoid needing warning layout logic in two places, could we instead have ScreenFrameWithFeed wrap the children content in a container who's max height is computed there? That way this can still assume 100%. My main concern is that if ScreenFrameWithFeed has other children than ScreenContentFrame, those other children will need this logic too.

brschwalm replied:
Ok, after taking another look, I was actually able to add it to Home Screen (when not Mobile) so it's within the content frame.  This eliminates the need for the layout logic in ScreenContentFrame, and also removes the need for the hook completely.  See if you think this looks better.

jmrossy commented on src/features/home/HomeScreenWarnings.tsx:
```diff
@@ -0,0 +1,73 @@
+import { Color } from 'src/styles/Color'
+import { fromWei } from 'src/utils/amount'
+
+const HIGH_VALUE_THRESHOLD = 25
```
Can you please move this to consts.ts

jmrossy commented on src/features/home/HomeScreenWarnings.tsx:
```diff
@@ -0,0 +1,73 @@
+export function useHomeScreenWarnings() {
+  const warning = useSelector(selectHomeScreenWarnings)
+  const location = useLocation()
+  //ScreenFrameWithFeed for desktop doesn't differentiate between home and not home, so need to check the path
```
It doesn't cause we didn't need to when I wrote that bit but it certainly can if it would help keep this stuff clearner. I'm fine either way

brschwalm replied:
Per above comment, I reorganized, so this is no longer necessary.

brschwalm commented:
Addressed the changes above, merged in latest from master, plus added the Tooltip component, and updated HelpIcon to use it.  There are several examples of tooltip stuff in the /dev/modals test form - if we want to get fancy, we can add any kind of content to the tooltip now.
